### DISCUSSION TITLE
How to make the setup look...neat?

### DISCUSSION THREAD

**Discussion by sapozd** [aqualinkd/AqualinkD#349] [General]
I connected the raspberry pi to my indoor, wall-mounted keypad. My control panel outside is too far for wifi to reach.

I plan to eventually hook the power up to the RS485 wiring. But even with that, using the recommended RS485 adapter there are just too many wires. My RS8 control panel hangs half open because there is no room for the both the raspberry pi and the wires.

I know that it's doable to fit it all inside the keypad, but I'm not creative enough to figure out how. Has anyone already done this and can provide some guidance on the tools needed to accomplish this?

**Comment by Tim-K5AC** [comment]
I replaced my All Touch RS-4 Keypad that was on the wall with the Raspberry Pi and a 10" touchscreen using the Web browser in Kiosk mode.

**Comment by sfeakes** [comment]
There are some pic's here.
https://github.com/sfeakes/AqualinkD/issues/19

You can cut / shorten the wire on the adapter.  There are also plenty of other adapters many smaller that will work. 
If you can solder, you can also get a SPI2RS485 board, those are usually a lot smaller. You then need to wire/solder a few pins directly to the GPIO of the raspberry.
I keep meaning to make a 3D model (than can easily be 3D printed) of a deeper keypad backing plate to make fitting the raspberry easier.
Another option would be to put a regular wall socket behind the keypad and put the pi in there.
